docs/multi-az-control-plane.md

```diff
@@ -0,0 +1,42 @@
+# Multi-AZ Control Plane
+
+## Overview
+
+By default, the control plane of a workload cluster created by CAPA will not span multiple availability zones (AZs) (also referred to as "failure domains"), even when using multiple control plane nodes. This is because CAPA will, by default, only create resources in the first discovered AZ of a region. However, if CAPA is specifically configured to create resources in multiple AZs, then control plane nodes will be automatically distributed across multiple AZs. This document explains how to instruct CAPA to create resources in multiple AZs.
+
+## Configuring CAPA to Use Multiple AZs
+
+To explicitly instruct CAPA to create resources in multiple AZs, users must add a `networkSpec` object to the AWSCluster specification. Here is an example `networkSpec` that creates resources across three AZs in the "us-west-2" region:
+
+```yaml
+spec:
+  networkSpec:
+    vpc:
+      cidrBlock: 10.50.0.0/16
+    subnets:
+    - availabilityZone: us-west-2a
+      cidrBlock: 10.50.0.0/20
+      isPublic: true
+    - availabilityZone: us-west-2a
+      cidrBlock: 10.50.16.0/20
+    - availabilityZone: us-west-2b
+      cidrBlock: 10.50.32.0/20
+      isPublic: true
+    - availabilityZone: us-west-2b
+      cidrBlock: 10.50.48.0/20
+    - availabilityZone: us-west-2c
+      cidrBlock: 10.50.64.0/20
+      isPublic: true
+    - availabilityZone: us-west-2c
+      cidrBlock: 10.50.80.0/20
+```
+
+Specifying the CIDR block alone for the VPC is not enough; users must also supply a list of subnets that provides the desired AZ, the CIDR for the subnet, and whether the subnet is public (has a route to an Internet gateway) or is private (does not have a route to an Internet gateway).
+
+Note that CAPA insists that there must be a public subnet (and associated Internet gateway) in order to place a control plane node in that AZ, even if the control plane load balancer is configured to be an internal load balancer. Therefore, for every AZ where a control plane node should be placed, the `networkSpec` object must define both a public and private subnet.
```

Comment:
We should probably just fix CAPA rather than having to document around this every time, but IIRC we check whether there _are_ public subnets, but then place the control plane machines in private subnets if the configured load balancer is `internal`.

Reply:
We should create a new issue if we don't have this already, and is ok to mark as a caveat in the doc at the present time?.

Reply:
Good call on opening the issue: https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/1696

For now, I'm thinking something like this:

```suggestion
Note that CAPA insists that there must be a public subnet (and associated Internet gateway), even if no public load balancer is requested for the control plane. Therefore, for every AZ where a control plane node should be placed, the `networkSpec` object must define both a public and private subnet.
```

As I understand it, CAPA will always place the nodes in private subnets, but depending on the load balancer's scheme may look for a public subnet to associate with the ELB itself.

Reply:
I've updated the wording as suggested in the latest commit (6ae70d0).